lib/mayaUsd/render/vp2RenderDelegate/proxyRenderDelegate.cpp

```diff
@@ -229,18 +226,40 @@ bool ProxyRenderDelegate::requiresUpdate(const MSubSceneContainer& container, co
     return true;
 }
 
+void ProxyRenderDelegate::_ClearRenderDelegate()
+{
+    delete _sceneDelegate;
```

Comment:
Instead of using the plain pointers as data member, a better way is to use unique_ptr.

Then this manual cleanup can be avoided and data members in the proxyRenderDelegate class can be declared consistently.